libs/blockscout-auth/src/lib.rs

```diff
@@ -0,0 +1,140 @@
+use cookie::Cookie;
+use serde::Serialize;
+use std::collections::HashMap;
+use thiserror::Error;
+use tonic::{codegen::http::header::COOKIE, metadata::MetadataMap, Request};
+
+const JWT_TOKEN_NAME: &str = "_explorer_key";
+const CSRF_TOKEN_NAME: &str = "_csrf_token";
+
+#[derive(Debug)]
+pub struct AuthSuccess {
+    pub user_id: String,
+}
+
+#[derive(Error, Debug, PartialEq, Eq)]
+pub enum Error {
+    #[error("invalid data: {0}")]
+    InvalidData(String),
```

Comment:
I suggest to create two errors instead `InvalidJwtToken` and `InvalidCsrfToken`, so that using crates could differentiate between those cases if would be needed.